addons/format_lame.c

```diff
@@ -0,0 +1,522 @@
+/*
+ * Asterisk -- An open source telephony toolkit.
+ *
+ * MP3 write support using LAME
+ * added by Naveen Albert <asterisk@phreaknet.org>
+ * and sponsored by Marfox Ltd.
+ *
+ * MP3 read support using LAME
+ * added by Sperl Viktor <viktike32@gmail.com>
+ *
+ * See http://www.asterisk.org for more information about
+ * the Asterisk project. Please do not directly contact
+ * any of the maintainers of this project for assistance;
+ * the project provides a web site, mailing lists and IRC
+ * channels for your use.
+ *
+ * This program is free software, distributed under the terms of
+ * the GNU General Public License Version 2. See the LICENSE file
+ * at the top of the source tree.
+ */
+
+/*!
+ * \file
+ * \brief MP3 Format including encoder and decoder
+ * \ingroup formats
+ */
+
+/*** MODULEINFO
+	<depend>lame</depend>
+	<conflict>format_mp3</conflict>
+	<defaultenabled>no</defaultenabled>
+	<support_level>extended</support_level>
+ ***/
+
+#include "asterisk.h"
+
+#include <lame/lame.h>
+
+#include "asterisk/module.h"
+#include "asterisk/mod_format.h"
+#include "asterisk/logger.h"
+#include "asterisk/format_cache.h"
+
+#define SLIN_BUFLEN 320
+#define DECORDER_OUTLEN 32768
+#define BITS 2
+
+struct lame_private {
+	lame_global_flags * lgfp;
+	hip_global_flags * hgfp;
+
+	int sample_rate;
+
+	/* decoder buffer */
+	short int doutput[DECORDER_OUTLEN *  1024];
+	long doffset;
+	int dsamples;
+
+	/* decoder positions */
+	int total_bytes_decoded, total_bytes_compressed;
+
+	/* counters */
+	unsigned int decoder_counter, encoder_counter;
+};
+
+static void ast_lame_error(const char *format, va_list ap)
```

Comment:
Don't use a `ast_` prefix unless you are exporting the symbol (which you are not).

This seems to apply to every function in this file, but I am only commenting on it once.

Reply:
See below. It's mp3lame_ now.